File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.java
class Solution {
    public int characterReplacement(String s, int k) {
        int maxLength = 0;
        for(int i=0;i<s.length();i++){
            Map<Character, Integer> map = new HashMap<>();
            map.put(s.charAt(i), 1);
            for(int j=i+1;j<s.length();j++){
                map.put(s.charAt(j), map.getOrDefault(s.charAt(j), 0)+1);
                int mostChar = Collections.max(map.values());
                if( (j-i+1 - mostChar)>k ){
                    break;
                }
                maxLength = Math.max(maxLength, j-i+1);
            }
        }
        return maxLength;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)==0:return 0
        if len(s)==1:return 1
        mx = 0
        l, r = 0, 1
        st = set()
        st.add(s[l])
        while r<len(s):
            if s[r] not in st:
                st.add(s[r])
                r+=1
                mx = max(mx, r-l)
            else:
                l=r
                st = set()
                st.add(l)
        
        return mx